pkg/kubectl/resource_printer.go

```diff
@@ -84,15 +86,15 @@ func GetPrinter(format, formatArgument string) (ResourcePrinter, bool, error) {
 // ResourcePrinter is an interface that knows how to print runtime objects.
 type ResourcePrinter interface {
 	// Print receives a runtime object, formats it and prints it to a writer.
-	PrintObj(runtime.Object, io.Writer) error
+	PrintObj(runtime.Object, *watch.EventType, io.Writer) error
 }
 
 // ResourcePrinterFunc is a function that can print objects
-type ResourcePrinterFunc func(runtime.Object, io.Writer) error
+type ResourcePrinterFunc func(runtime.Object, *watch.EventType, io.Writer) error
```

Comment:
On Jul 29, 2015, at 6:28 PM, Marcin Wielgus <notifications@github.com>
wrote:

In pkg/kubectl/resource_printer.go
<https://github.com/GoogleCloudPlatform/kubernetes/pull/11440#discussion_r35819489>
:

>  }
>
>  // ResourcePrinterFunc is a function that can print objects
> -type ResourcePrinterFunc func(runtime.Object, io.Writer) error
> +type ResourcePrinterFunc func(runtime.Object, *watch.EventType, io.Writer) error

I don't understand your function signature. Do you want
HumanReadablePrinter to be the only printer that supports events? What
should happen if a person calls kubectl get pods -o=json -w?


I don't think there's anything yaml or json need to do to support events -
they are already generic (although there may be some annoyances)

There's only one printer that needs to support events directly, and it's
used by -w for human readable output.  I was proposing that printer be the
outer wrapper, unwrapping the object from the event, invoking a
HumanReadablePrinter on the nested object, then prepending a column.  That
should satisfy keeping the normal HRP from knowing about events.


—
Reply to this email directly or view it on GitHub
<https://github.com/GoogleCloudPlatform/kubernetes/pull/11440/files#r35819489>
.